### PR TITLE
414 informers use best match

### LINKED ISSUE #414: Informers still use exact match to find workloads/deliverables for supply chains/deliveries
### Bug description:
In #359 we changed the behavior of supply chains and deliveries to match workloads and deliverables based on a "best match" of labels. This change was not made in the watchers, though, and so when a change happens to a supply chain or delivery, it still looks for the relevant workloads and deliverables by matching labels exactly (see [here](https://github.com/vmware-tanzu/cartographer/blob/main/pkg/registrar/map_functions.go#L116) and [here](https://github.com/vmware-tanzu/cartographer/blob/main/pkg/registrar/map_functions.go#L148), respectively).

### Steps to reproduce:
Please provide a set of steps to reproduce the behavior (example):
1. Apply supply chain A with label X and supply chain B with labels X and Y
2. Apply a workload that has labels X and Y
3. Make a change to supply chain A

### Expected behavior:
The workload should not re-reconcile because it matches supply chain B better

### Actual behavior:
The workload does re-reconcile

### PR DESCRIPTION
**Pull request by martyspiewak**
Closes #414

### CODE REVIEW

File: pkg/registrar/map_functions_test.go [hunk 1]
```diff
@@ -471,6 +506,26 @@ var _ = Describe("MapFunctions", func() {
...
+							sc := &v1alpha1.ClusterSupplyChain{
+								Spec: v1alpha1.SupplyChainSpec{
+									Selector: map[string]string{
+										"myLabel":      "myLabelsValue",
+										"myOtherLabel": "myOtherLabelsValue",
+									},
+								},
+							}
+
+							clientObjects = []client.Object{workload, sc}
```
Comment (waciumawanjohi) [inline pkg/registrar/map_functions_test.go]: Pretty sure this should be
```
clientObjects = []client.Object{workload, sc, clusterSupplyChain}
```

removing `sc` should make the test fail, but in the current setup it does not.

File: pkg/registrar/map_functions.go [hunk 2]
```diff
@@ -127,17 +129,45 @@ func (mapper *Mapper) ClusterSupplyChainToWorkloadRequests(object client.Object)
...
+	scList := &v1alpha1.ClusterSupplyChainList{}
+	err = mapper.Client.List(context.TODO(), scList)
+	if err != nil {
+		mapper.Logger.Error(err, "cluster supply chain to workloads: client list supply chains")
+		return nil, err
+	}
+
+	var selectorGetters []repository.SelectorGetter
+	for _, item := range scList.Items {
+		item := item
```
Comment (waciumawanjohi) [inline pkg/registrar/map_functions.go:147]: Not blocking:
Is this just to create a copy of the object? Would `persistentItem := item.DeepCopy()` suit? Would it be more grokkable?

**Review by waciumawanjohi — CHANGES_REQUESTED** [review]
Just the one change to fix the test. Otherwise, lgtm!

File: pkg/registrar/map_functions.go (hunk 2 above)
Comment (martyspiewak) [reply pkg/registrar/map_functions.go:147]: I think that would work too but I was trying to be consistent with the way the controller finds the best matching SC (see [here](https://github.com/vmware-tanzu/cartographer/blob/main/pkg/repository/repository.go#L286-L290))

File: pkg/registrar/map_functions_test.go (hunk 1 above)
Comment (martyspiewak) [reply pkg/registrar/map_functions_test.go]: Good catch thanks!

**Review by waciumawanjohi — APPROVED** [review]

**Pull request merged**